tls/s2n_prf.c

```diff
@@ -672,8 +672,21 @@ static int s2n_sslv3_server_finished(struct s2n_connection *conn)
     return s2n_sslv3_finished(conn, prefix, &conn->handshake.hashes->hash_workspace, conn->handshake.server_finished);
 }
 
-int s2n_prf_client_finished(struct s2n_connection *conn)
+S2N_RESULT s2n_prf_client_finished(struct s2n_connection *conn)
 {
+    RESULT_ENSURE_REF(conn);
+
+    if (conn->mode == S2N_SERVER && !conn->npn_negotiated
+        && s2n_conn_get_current_message_type(conn) == CLIENT_FINISHED) {
+        return S2N_RESULT_OK;
+    }
```

Comment:
This method definitely needs some comments. I know we've had some fun conversations about this. What does the handshake hash need to include? How does npn change that? (How does that violate the RFC :( )